### ISSUE TITLE
Ready to publish?

### ISSUE THREAD

**Issue by joneshf** [paulyoung/purescript-corefn#44]
Heya! This package looks great! Any interest in publishing it for others to use?

**Comment by paulyoung** [comment]
Yeah, I've been thinking about it now that #42 has been merged and I've validated that it works as expected so far in [PureSwift](https://github.com/paulyoung/pureswift).

The only thing is that it's now only compatible with the corefn output of 0.12 so I was wondering if I should wait for that to be released.

What do you think?

**Comment by joneshf** [comment]
Actually, that's maybe a big ask. Dunno what kind of maintenance you want to do on this library. Anyone can follow suit like you did on `PureSwift` and reference a commit.

**Comment by paulyoung** [comment]
#42 was pretty much a straight port from Haskell and is fully featured whereas the work before that which was compatible with 0.11 only just about supported lambda calculus from what I remember.

The big motivation for switching to the 0.12 corefn output is that it's the only way (so far as I could tell) to support FFI in alternative backends. This issue I opened explains it a little bit: https://github.com/purescript/purescript/issues/3162

That, along with the hope that 0.12 might be released soon means I wasn't planning on supporting the 0.11 output.

FWIW, the branch of PureSwift that's using the new format is [paulyoung/pswc](https://github.com/paulyoung/pureswift/tree/paulyoung/pswc).

**Comment by joneshf** [comment]
Sorry, I know I said I would hold out, but I kinda just went ahead and did it since I don't know when 0.12 is coming out: https://github.com/paulyoung/purescript-corefn/compare/646c996ac179b3f60f5abe40aa32ca005809e2c0...joneshf:master. If you'd like to support 0.11.7, I can open a PR if you make a branch at 646c996ac179b3f60f5abe40aa32ca005809e2c0.

As far as finding FFI files, that's unfortunate. Seems like there's two common cases (as you pointed out): bower and psc-package. Probably can support those two cases explicitly and call it a day.

**Comment by paulyoung** [comment]
I just created a `0.11` branch 🙂

**Comment by paulyoung** [comment]
I think there’s still an issue with FFI in that you know the module name but not the package name.

**Comment by paulyoung** [comment]
For example, you have `Control.Monad.Eff.Console` but don’t know that it belongs to `purescript-console`

**Comment by joneshf** [comment]
Oh right. Hmm, that is unfortunate.

**Comment by joneshf** [comment]
Wait, could you do some processing to figure it out?

Again, assuming the two package managers, dip into the respective directories (`bower_components` or `.psc-package`) and map modules to packages. Anything that's not in that mapping has to be from the local source. Once you have a module, look it up in the mapping and fallback to the local source if it doesn't exist.

**Comment by paulyoung** [comment]
That sounds like it could work!

**Comment by paulyoung** [comment]
Should I make a release for the 0.11 stuff?

**Comment by joneshf** [comment]
Lemme throw in a functor instance, then yes!

**Comment by paulyoung** [comment]
I’m not sure how to approach versioning here. For example, if I make a v1.0.0 release for 0.11 output and then later make a v2.0.0 release for 0.12 output, what would happen when the 0.11 branch needs to make a breaking change?

**Comment by joneshf** [comment]
3.0.0? It's a breaking change for both 1.0.0 and 2.0.0.

I dunno, I'm pretty up in the air about versions. I'm probably not the best person to ask for advice :). Also, I'm fine with using a commit SHA for now if you don't want to think about it.

**Comment by paulyoung** [comment]
A somewhat radical proposal might be to have something like `CoreFn.0.11.Module` and `CoreFn.0.12.Module` 😯
